src/metrics.js

```diff
@@ -0,0 +1,126 @@
+            var emitParams = {
+                type: 'errorReport',
+                subtype: 'default',
+                doc: {
+                    input: param.input,
+                    date: param.date
+                },
+                _cb: param._cb
+            }
+            self._eventBus.emitEvent(eventObj.DATABASE_ADD, emitParams);
+        });
+
+        self._eventBus.addEventListener(eventObj.COMMAND_REPORT_ADD, this, function(param) {
+            var inputIndividualData = {
+                date: moment().format('x'),
+                input: param.inputParams
+            }
+            var emitParams = {
+                type: 'commandReport',
+                fetchParams: {
+                    'command': param.command
+                },
+                _cb: function(err, docs) {
+                    if (err) {
+                        console.log(err);
+                    }
+                    if (Object.keys(docs).length == 0) {
+                        var addEmitParams = {
+                            type: 'commandReport',
+                            subtype: 'default',
+                            doc: {
+                                command: param.command,
+                                totalUsage: 1,
+                                individualData: [inputIndividualData]
+                            },
+                            _cb: param._cb
+                        }
+                        self._eventBus.emitEvent(eventObj.DATABASE_ADD, addEmitParams);
+                    } else {
+                        var updateEmitParams = {
+                            type: 'commandReport',
+                            params: {
+                                'command': param.command
+                            },
+                            updates: {
+                                $push: {
+                                    'individualData': paramIndividualData
+                                }
+                            },
+                            _cb: param._cb
+                        }
+                        self._eventBus.emitEvent(eventObj.DATABASE_UPDATE, updateEmitParams);
+                    }
+                }
+            }
+            self._eventBus.emitEvent(eventObj.DATABASE_FETCH, emitParams);
+        });
+
+        self._eventBus.addEventListener(eventObj.DAILY_REPORT_ADD, this, function(param) {
+            var inputIndividualData = {
+                command: param.command,
+                date: moment().startOf('d').format('x'),
+                time: moment().format('x')
+            }
+
+            var emitParams = {
+                type: 'dailyReport',
+                fetchParams: {
+                    'date': inputIndividualData.date
+                },
+                _cb: function(err, docs) {
+                    if (err) {
+                        param._cb(err);
+                    }
+                    if (Object.keys(docs).length == 0) {
```

Comment:
This is just checking if docs doesn't exist right? Condensing this to if (!docs) should be enough I believe